src/steamship_langchain/file_loaders/github.py

```diff
@@ -0,0 +1,82 @@
+"""Import files from a GitHub repository into Steamship workspace."""
+import tempfile
+from pathlib import PurePath
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field
+from steamship import File, Steamship, Tag
+from steamship.data import TagKind, TagValueKey
+from steamship.data.tags.tag_constants import ProvenanceTag
+
+from steamship_langchain.file_loaders.directory import DirectoryLoader
+from steamship_langchain.file_loaders.text import TextFileLoader
+
+
+class GitHubRepositoryLoader(BaseModel):
```

Comment:
Is this supposed to inherit from `BaseFileLoader`?

Reply:
Good question. The short-term answer is not at the moment, because it has a slightly different interface. I think a second pass when more importers are added will be a good chance to refactor. It felt wrong to have the `path` be in the instance params and not in the call to `load`, but having in `load` means that GH wouldn't really use it. Open to better structure for sure. I expect this will evolve as more importers are added.